nav2_velocity_smoother/src/velocity_smoother.cpp

```diff
@@ -267,35 +267,72 @@ void VelocitySmoother::smootherTimer()
     command_->angular.z = 0.0;
   }
 
+  double accel_eta = 1.0;
+  double decel_eta = 1.0;
   double eta = 1.0;
+
+  bool x_accelerates = true;
+  bool y_accelerates = true;
+  bool angular_accelerates = true;
+
   if (scale_velocities_) {
     double curr_eta = -1.0;
 
     curr_eta = findEtaConstraint(
       current_.linear.x, command_->linear.x, max_accels_[0], max_decels_[0]);
-    if (curr_eta > 0.0 && std::fabs(1.0 - curr_eta) > std::fabs(1.0 - eta)) {
-      eta = curr_eta;
+
+    if (curr_eta > 0.0) {
+      if (std::fabs(1.0 - curr_eta) > std::fabs(1.0 - eta)) {
```

Comment:
What is the purpose of that `1.0`?

Reply:
Hmm.. I'm not the author of that particular condition, it appears in diff because I brought out the curr_eta > 0 condition into an outer if. 
But I think it's because smaller eta is supposed to be considered to be more significant.

Reply:
If that, I wonder why they did not prefer to write `if (std::fabs(curr_eta) < std::fabs(eta))`.